haystack/document_stores/__init__.py

```diff
@@ -7,7 +10,22 @@
 from haystack.document_stores.es_converter import elasticsearch_index_to_document_store
 from haystack.document_stores.es_converter import open_search_index_to_document_store
 
-from haystack.document_stores.elasticsearch import ElasticsearchDocumentStore
+try:
+    # Use appropriate ElasticsearchDocumentStore depending on ES client version
+    from elasticsearch import __version__ as ES_VERSION
+
+    if ES_VERSION[0] == 7:
+        from haystack.document_stores.elasticsearch7 import ElasticsearchDocumentStore  # type: ignore  # pylint: disable=reimported,ungrouped-imports
+    elif ES_VERSION[0] == 8:
+        if not typing.TYPE_CHECKING:
+            from haystack.document_stores.elasticsearch8 import ElasticsearchDocumentStore  # type: ignore  # pylint: disable=reimported,ungrouped-imports
+    else:
+        # Use Elasticsearch 7 as default
+        from haystack.document_stores.elasticsearch7 import ElasticsearchDocumentStore
+except (ModuleNotFoundError, ImportError):
+    # Import ES 7 as default if ES is not installed to raise the error message that elasticsearch extra is needed
+    from haystack.document_stores.elasticsearch7 import ElasticsearchDocumentStore
+
```

Comment:
Can we maybe use `lazy_import` here?

Also why disable import of version 8 if type checking is running?

Reply:
Just noticed that we're actually importing `lazy_import` and doing nothing with that. 😕